website/docs/fundamentals/managed-node-groups/spot/index.md

```diff
@@ -3,21 +3,151 @@ title: Spot instances
+ip-10-42-10-232.us-west-2.compute.internal   Ready    <none>   113m   v1.23.15-eks-49d8fe8   ON_DEMAND      managed-system-20230605211737831800000026
+ip-10-42-10-96.us-west-2.compute.internal    Ready    <none>   113m   v1.23.15-eks-49d8fe8   ON_DEMAND      managed-ondemand-20230605211738568600000028
+ip-10-42-11-17.us-west-2.compute.internal    Ready    <none>   78s    v1.23.17-eks-0a21954   SPOT           managed-spot
+ip-10-42-12-234.us-west-2.compute.internal   Ready    <none>   77s    v1.23.17-eks-0a21954   SPOT           managed-spot
+ip-10-42-12-45.us-west-2.compute.internal    Ready    <none>   113m   v1.23.15-eks-49d8fe8   ON_DEMAND      managed-ondemand-20230605211738568600000028
+
+```
+From the above output you can see that now we have two managed nodegroups available. Next, let’s deploy the “Sample Retail Store” app and leverage [nodeSelector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) as node selection constraint to deploy app on top of the spot instances instead of On-Demand. Using the nodeSelector field, you can specify key-value pairs to filter and select which nodes in the cluster should run particular pods. It is used to define constraints for pod placement based on node labels. 
+
+Now because our existing manifest `deployment.yaml` located under `/workspace/manifests/catalog`  does not have `nodeSelector` as an attribute, we will use [kustomize](https://www.eksworkshop.com/docs/introduction/kustomize/) to modify the existing resource configuration without directly modifying the original manifests. 
+
+```kustomization
+fundamentals/mng/spot/deployment.yaml
+Deployment/catalog
+```
+
+Next, Deploy the app.
+
+```bash
+$ kubectl apply -k /workspace/modules/fundamentals/mng/spot
+
+namespace/catalog created
+serviceaccount/catalog created
+configmap/catalog created
+secret/catalog-db created
+service/catalog created
+service/catalog-mysql created
+deployment.apps/catalog created
+statefulset.apps/catalog-mysql created
+```
+
+To make sure your app got deployed successfully, We will use `kubectl rollout status` command that allows you to check the status of a deployment in Kubernetes. It provides information about the progress of the rollout and the current state of the associated resources.
+```bash
+$ kubectl rollout status deployment/catalog -n catalog --timeout=5m
+```
+Once the application get’s deployed successfully, The Final step would be to check whether the app got deployed on `SPOT` instances, To do that run below script on your terminal.
+
+```bash
+$ for pod in $(kubectl get pods -l app.kubernetes.io/name=catalog -n $CATALOG_RDS_DATABASE_NAME  | awk 'NR>1 {print $1}'); do if [ "$pod" != "NAME" ]; then node=$(kubectl get pod $pod -o json -n $CATALOG_RDS_DATABASE_NAME | jq -r '.spec.nodeName'); node_group=$(kubectl get node $node -o json | jq -r '.metadata.labels."eks.amazonaws.com/capacityType"'); echo "$pod | $node | $node_group"; fi; done
```

Comment:
Instead of a custom command, can we do a combination of `$ kubectl get nodes -l eks.amazonaws.com/capacityType=ON_DEMAND` like above, followed by a `kubectl get pods`:

`kubectl get pod -o=custom-columns=NAME:.metadata.name,STATUS:.status.phase,NODE:.spec.nodeName --all-namespaces`

User can connect the dots.